### PR TITLE
Add Unit test structure for Envoy mgw oasparser package 

### LINKED ISSUE #1257: OpenApi 3 VendorExtenstible are not unmarshelling correctly in Envoy mgw
### Description:

As $Subject in envoy mgw, when unmarshelling the openapi 3 yaml files, it's not unmarshelling vendor extensions correctly and it remain as the bytes.

### Steps to reproduce:
Provide the openApi3 definition as the api defintion  with vendor extensions  (x-wso2-prodution-endpoints, x-wso2-basepath)  and generate the envoy artifacts

### Affected Product Version:

Envoy mgw

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by chashikajw**
### Purpose

This will add a unite test for oasparser package in envoy mgw.  

**Case 1**:  
Situations where you can unit test everything just from the external (Usual scenario)
- So for package **xyz**, put your test files in package **xyz_test** (same folder).
- Test file name should be **file_test**.

**Case 2:** 
Situations where you cannot unit test everything just from the external( having private variables,struct inside the package). So when you really need to test package internals.

- Use the same package as the code under test (e.g. xyz)
- Name your test file with the **_internal_test.go** suffix (so xyz_internal_test.go) to make it explicit and clear that you know these tests are more brittle.

**Test resources**
If your test cases require test resources such as swagger files, apis, create a separate directory called **testdata** inside the same directory and include the resources.

**Test cases**

- All the test cases should start with the Prefix Test and after the Test next word first letter should be a capital letter and the parameter should be t *testing.T.
Ex: Test**G**etresources (t *testing.T)

- Use a test struct if u have more test cases
`Ex:  type somethingTestItem struct {
        input   string
        expectedResult  string
        message string
    }
`

Ex:  Full test case
```
func TestSomething(t *testing.T) {
	type somethingTestItem struct {
		input string
		result string
		message string
	}
	dataItems := []SomethingTestItem {
		{
			input: "input data",
			result: "Expected result"
			message: "case 1",  
		},
		
	}
	for _, item := range dataItems{
		actual := getHostandBasepathandPort(item.input)

		assert.Equal(t, item.result, actual , item.message)
	}
}
```

### Issues

Fixes #1257 #1255

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: internal/pkg/oasparser/models/apiDefinition/openApi.go [hunk 1]
```diff
@@ -34,72 +35,71 @@ func (swagger *MgwSwagger) SetInfoOpenApi(swagger3 openapi3.Swagger) {
...
+				resource = setOperationOpenApi(path, "put", pathItem.Put)
+			} else if pathItem.Delete != nil {
+				resource = setOperationOpenApi(path, "delete", pathItem.Delete)
+			} else if pathItem.Head != nil {
+				resource = setOperationOpenApi(path, "head", pathItem.Head)
+			} else if pathItem.Patch != nil {
+				resource = setOperationOpenApi(path, "patch", pathItem.Patch)
+			} else {
+				//resource = setOperation(contxt,"get",pathItem.Get)
+			}
```
Comment (chathuranga95) [inline internal/pkg/oasparser/models/apiDefinition/openApi.go:86]: If the commented out line is not needed, please remove the empty `else` block

File: internal/pkg/oasparser/models/apiDefinition/swagger.go [hunk 2]
```diff
@@ -35,45 +35,51 @@ func (swagger *MgwSwagger) SetInfoSwagger(swagger2 spec.Swagger) {
...
+				resource = setOperationSwagger(path, "put", pathItem.Put)
+			} else if pathItem.Delete != nil {
+				resource = setOperationSwagger(path, "delete", pathItem.Delete)
+			} else if pathItem.Head != nil {
+				resource = setOperationSwagger(path, "head", pathItem.Head)
+			} else if pathItem.Patch != nil {
+				resource = setOperationSwagger(path, "patch", pathItem.Patch)
+			} else {
+				//resource = setOperation(contxt,"get",pathItem.Get)
+			}
```
Comment (chathuranga95) [inline internal/pkg/oasparser/models/apiDefinition/swagger.go:62]: If the commented out line is not needed, please remove the empty `else` block

File: internal/pkg/oasparser/swaggerOperator/swaggerOperator.go [hunk 3]
```diff
@@ -32,65 +32,73 @@ func GenerateMgwSwagger(location string) ([]apiDefinition.MgwSwagger, error) {
...
+}
+
+func GetMgwSwagger(apiContent []byte) apiDefinition.MgwSwagger {
+	var mgwSwagger apiDefinition.MgwSwagger
 
-		swaggerVerison, err := utills.FindSwaggerVersion(apiJsn)
+	apiJsn, err := utills.ToJSON(apiContent)
+	if err != nil {
+		//log.Fatal("Error converting api file to json:", err)
+	}
```
Comment (chathuranga95) [inline internal/pkg/oasparser/swaggerOperator/swaggerOperator.go:68]: Empty `if` block.

File: internal/pkg/oasparser/models/apiDefinition/openApi.go (hunk 1 above)
Comment (chashikajw) [reply internal/pkg/oasparser/models/apiDefinition/openApi.go:86]: This is already removed

File: internal/pkg/oasparser/models/apiDefinition/swagger.go (hunk 2 above)
Comment (chashikajw) [reply internal/pkg/oasparser/models/apiDefinition/swagger.go:62]: This is already fixed

File: internal/pkg/oasparser/swaggerOperator/swaggerOperator.go (hunk 3 above)
Comment (chashikajw) [reply internal/pkg/oasparser/swaggerOperator/swaggerOperator.go:68]: This is already fixed

File: internal/pkg/oasparser/utills/openApiUtills_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+func TestFindSwaggerVersion(t *testing.T) {
+
+	type findSwaggerVersionTestItem struct {
+		inputSwagger   string
+		result  string
+		message string
+	}
+
+	dataItems := []findSwaggerVersionTestItem {
+		{
```
[the remark this answers is not in the document]
Comment (chashikajw) [reply internal/pkg/oasparser/utills/openApiUtills_test.go:34]: This is fixed

**Pull request merged**